pkg/datasource.go

```diff
@@ -503,105 +507,133 @@ func (o *OCIDatasource) processLogRecords(ctx context.Context, searchLogsReq Gra
+											"panelId", queryPanelId, "refId", queryRefId, "type", fmt.Sprintf("%T", value))
+										logData = "UNKNOWN"
+									}
+									fieldDefn.Values.([]*string)[indexCountPag] = &logData
+
+									// Skip the subject field since it seems to always be an empty string
+									// For all other keys treat them generically as string type
+								} else if key != LogSearchResultsField_Subject {
+									var stringFieldValue string
+									fieldDefn = nil
+
+									if stringFieldValue, ok = value.(string); ok {
+										// If the field value is non-zero length string then proceed to get/create the data
+										// field definition. But if the field value is a zero length string then skip
+										// creating the data field definition, this is to avoid creating a data field for a
+										// log record field that is always empty.
+										if len(stringFieldValue) > 0 {
+											fieldDefn = o.getCreateDataFieldElemsForField(mFieldDefns, numMaxResults,
+												key, key, ValueType_String)
+										}
+									} else {
+										o.logger.Debug("Error parsing string field value", "panelId", queryPanelId,
+											"refId", queryRefId, "key", key, "value", value)
+										fieldDefn = o.getCreateDataFieldElemsForField(mFieldDefns, numMaxResults,
+											key, key, ValueType_String)
+										stringFieldValue = "UNKNOWN"
+									}
+									if fieldDefn != nil {
+										fieldDefn.Values.([]*string)[indexCountPag] = &stringFieldValue
+									}
+								} // endif key name
+							} // for each field key in the logContent field
+
+						} else {
+							o.logger.Debug("Unable to get logContent map", "panelId", queryPanelId,
+								"refId", queryRefId, "row", rowCount)
+						}
 					} else {
-						o.logger.Debug("Unable to get logContent map", "panelId", queryPanelId,
-							"refId", queryRefId, "row", rowCount)
+						o.logger.Debug("Encountered log record without a logContent element",
+							"panelId", queryPanelId, "refId", queryRefId, "row", rowCount)
 					}
 				} else {
-					o.logger.Debug("Encountered log record without a logContent element",
-						"panelId", queryPanelId, "refId", queryRefId, "row", rowCount)
+					o.logger.Debug("Encountered row without a log record", "panelId", queryPanelId,
+						"refId", queryRefId, "row", rowCount)
 				}
-			} else {
-				o.logger.Debug("Encountered row without a log record", "panelId", queryPanelId,
-					"refId", queryRefId, "row", rowCount)
+				indexCountPag++
 			}
+
+		} else {
+			o.logger.Warn("Logging search query returned no results", "panelId", queryPanelId,
+				"refId", queryRefId)
+		}
+		if res.OpcNextPage != nil && numpage < MaxPagesToFetch {
+			// if there are more items in next page, fetch items from next page
+			request.Page = res.OpcNextPage
+			numpage++
+		} else {
+			o.logger.Debug("numpage :", "numpage", numpage)
+			o.logger.Debug("resultCountPag :", "resultCountPag", resultCountPag)
```

Comment:
Eliminate some of these debug messages or combine them. This and the next message are duplicative so I would suggest removing this one.

Reply:
ok, done